### DISCUSSION TITLE
Allow vsock and tcp connections

### DISCUSSION THREAD

**Discussion by voltamund** [neutrinolabs/xrdp#2077] [Q&A] [answered]
I want to use xrdp with enhanced session mode to a Hyper-V guest.
To do this I change the port in xrdp.ini like so
port=vsock://-1:3389

The drawback of this change is that I cannot use TCP to connect remotely (not from the Hyper-V host).
Is there a way to allow both connections? There will be only one connection at a time.

**Comment by matt335672** [comment]
**Edit** This is an old reply and completely unhelpful!

I've had a look at the code in `xrdp/xrdp_listen.c` and I'm pretty sure that xrdp can only listen on one transport at a time. The only exception to this is IPv4 + IPv6. 

The only thing I can suggest is to set up a proxy which listens on TCP and forwards the stream to the xrdp vsock. I don't have a Hyper-V hypervisor, so I can't try this though. `socat` version 1.7.4 looks like it should support this kind of thing but that's very new, so your guest may not have that easily available.

**Comment by shawn666liu** [comment, marked as answer]
port=vsock://-1:3389  tcp://:3389
https://c-nergy.be/blog/?p=17109
I searched the web and found that could work.
